File: backend/src/main/java/com/flowlink/execution/dto/PendingInputRequest.java
package com.flowlink.execution.dto;

import java.util.List;

/**
 * 사용자 입력 대기(INPUT) 노드에서 실행이 중단됐을 때, 브라우저가 띄울 입력 창 명세.
 * 사용자가 제출한 값이 노드 출력이 되어 다운스트림에서 {@code {{ key }}} 로 바인딩된다.
 */
public record PendingInputRequest(
        String nodeId,
        String nodeName,
        String msg,
        List<InputField> fields
) {
    public record InputField(String key, String label) {
    }
}

File: backend/src/main/kotlin/com/flowlink/execution/dto/PendingInputRequest.kt
package com.flowlink.execution.dto

/**
 * input(사용자 입력) 노드에서 실행이 중단됐을 때, 브라우저가 모달(input box)로 띄울 입력 명세.
 * 사용자가 confirm 하면 [ResumeRequest.formValues] 로 값이 돌아와 노드 출력이 된다.
 * [InputField.type] 은 값 파싱 힌트(string 기본 · number · boolean · json) — 브라우저가 파싱해 보낸다.
 */
data class PendingInputRequest(
    val nodeId: String?,
    val nodeName: String?,
    val message: String?,
    val fields: List<InputField>
) {
    data class InputField(val key: String?, val label: String?, val type: String?)
}
